virttest/utils_misc.py

```diff
@@ -4704,3 +4706,41 @@ def get_sosreport(path=None, session=None, remote_ip=None, remote_pwd=None,
         if session:
             session.close()
         return host_path
+
+
+def get_monitor_function(vm):
+    """
+    Get support function by function name
+    """
+    caller_name = inspect.stack()[1][3]
+    cmd = get_monitor_cmd(vm, caller_name.replace("_", "-"))
+    func_name = cmd.replace("-", "_")
+    return getattr(vm.monitor, func_name)
+
+
+def get_monitor_cmd(vm, cmd):
+    """
+    Wrapper to get monitor function
+    """
+    try:
+        vm.monitor.verify_supported_cmd(cmd)
+    except Exception as e:
+        logging.info(e.message)
+        cmd = "x-" + cmd
+        vm.monitor.verify_supported_cmd(cmd)
+    return cmd
```

Comment:
I think there are several issues regarding to the above two functions:

1. these are monitor related funtionalities so we should put them in `qemu_monitor`
2. we can have a simpler approach to handle this thing, for example:
```python
def get_monitor_function(vm):
    ...
    if hasattr(vm.monitor, caller_name):
        return getattr(vm.monitor, caller_name)
    # try the 'x' version
    return getattr(vm.monitor, "x_%s" % caller_name)
```
3. I do suggest passing the function name explicitly by the caller instead of calling `inspect.stack()[1][3]`, for example: 
```python
func = get_monitor_function(vm, 'device_add')
```
and that would make the naming of callers more flexible.